src/Player/PlayerHackKeeper.h

```diff
@@ -1,20 +1,77 @@
 #pragma once
 
-#include <al/Library/LiveActor/LiveActor.h>
+#include <math/seadMatrix.h>
+#include <container/seadPtrArray.h>
+
+namespace al {
+    class LiveActor;
+    class HitSensor;
+    class CollisionPartsFilterBase;
+} // namespace al
 
 class PlayerRecoverySafetyPoint;
 class HackCap;
 class PlayerInput;
 class PlayerDamageKeeper;
 class IPlayerModelChanger;
 class IUsePlayerHeightCheck;
+class HackObjInfo;
+class CapTargetInfo;
+class PlayerHackStartTexKeeper;
 
 class PlayerHackKeeper {
+private:
+    al::LiveActor* mParent;
+    HackCap* mHackCap;
+    PlayerRecoverySafetyPoint* mRecoverySafePoint;
+    void* field_18;
+    void* field_20;
+    PlayerInput* mInput;
+    sead::Matrix34f* field_30;
+    PlayerDamageKeeper* mDamageKeeper;
+    IPlayerModelChanger* mModelChanger;
+    IUsePlayerHeightCheck* mHeightCheck;
+    al::HitSensor* mParentBodySensor;
+    bool mIsPuppetable;
+    bool mIsCancellingHack;
+    bool mIsHackDemoStarted;
+    bool mIsPuppetable2;
+    bool mIsStartedHacking;
+    bool mIsHack;
+    bool mIsTookDamage;
+    al::CollisionPartsFilterBase* mCollisionFilter;
+    al::LiveActor* mHackActor;
+    al::HitSensor* mHackHitSensor;
+    HackObjInfo* mHackObjectInfo;
+    al::HitSensor* mStageStartActorSensor;
+    al::LiveActor* mStageStartActor;
+    CapTargetInfo* mStageStartCapTargetInfo;
+    PlayerHackStartTexKeeper* mHackStartTexKeeper;
+    al::LiveActor* mHackModel;
+    sead::PtrArray<void> mHackModelSlices;
```

Comment:
Put the real type in here, can be done as placeholder-struct

Reply:
i have no idea what the type is supposed to be... finding uses of it in ghidra doesn't really help deduce anything at all.

Reply:
How did you find a proper name for this entry then?

Reply:
gonna be honest i dont think i named this one, it might've been sanae since we share a ghidra project. but the PtrArray is allocated in `PlayerHackKeeper::createHackModel`. nowhere else where its used really helps figure out the type of the array that much.